pkg/canary/config_tracker.go

```diff
@@ -299,22 +299,38 @@ func (ct *ConfigTracker) CreatePrimaryConfigs(cd *flaggerv1.Canary, refs map[str
 		case ConfigRefMap:
 			config, err := ct.KubeClient.CoreV1().ConfigMaps(cd.Namespace).Get(context.TODO(), ref.Name, metav1.GetOptions{})
 			if err != nil {
-				return fmt.Errorf("configmap %s.%s get query failed : %w", ref.Name, cd.Name, err)
+				return fmt.Errorf("configmap %s.%s get query failed : %w", ref.Name, cd.Namespace, err)
 			}
 			primaryName := fmt.Sprintf("%s-primary", config.GetName())
+			ownerReferences := []metav1.OwnerReference{
+				*metav1.NewControllerRef(cd, schema.GroupVersionKind{
+					Group:   flaggerv1.SchemeGroupVersion.Group,
+					Version: flaggerv1.SchemeGroupVersion.Version,
+					Kind:    flaggerv1.CanaryKind,
+				}),
+			}
+
+			oldPrimary, err := ct.KubeClient.CoreV1().ConfigMaps(cd.Namespace).Get(context.TODO(), primaryName, metav1.GetOptions{})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("configmap %s.%s get query failed : %w", primaryName, cd.Namespace, err)
+				}
+			} else {
+				for _, ownerRef := range oldPrimary.OwnerReferences {
+					if ownerRef.Kind != flaggerv1.CanaryKind || ownerRef.Name != cd.Name {
+						ownerRef.Controller = new(bool)
```

Comment:
What's the reason for setting this to false?

Reply:
Only one of the owner references can be marked as controller (hard requirement in Kubernetes). The GC is not affected by which one is controller, and I could not find anything in Flagger using the controller attribute.